src/main/webapp/app/shared/user-settings/learner-profile/course-learner-profile.component.ts

```diff
@@ -0,0 +1,143 @@
+import { Component, OnInit, WritableSignal, inject, signal } from '@angular/core';
+import { EditableSliderComponent } from 'app/shared/editable-slider/editable-slider.component';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import { Course } from 'app/entities/course.model';
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { TranslateService } from '@ngx-translate/core';
+import { LearnerProfileApiService } from 'app/atlas/service/learner-profile-api.service';
+import { CourseLearnerProfileDTO } from 'app/entities/learner-profile.model';
+import { EditStateTransition } from 'app/shared/editable-slider/edit-process.component';
+import { AlertService, AlertType } from 'app/core/util/alert.service';
+import { TranslateDirective } from 'app/shared/language/translate.directive';
+
+@Component({
+    selector: 'jhi-course-learner-profile',
+    templateUrl: './course-learner-profile.component.html',
+    styleUrls: ['./course-learner-profile.component.scss'],
+    standalone: true,
+    imports: [EditableSliderComponent, TranslateDirective],
+})
+export class CourseLearnerProfileComponent implements OnInit {
+    private courseManagementService = inject(CourseManagementService);
+
+    courses: Course[];
+    courseLearnerProfiles: Record<number, CourseLearnerProfileDTO>;
+    activeCourse: number;
+
+    disabled = true;
+    aimForGradeOrBonus = signal<number>(0);
+    timeInvestment = signal<number>(0);
+    repetitionIntensity = signal<number>(0);
+
+    aimForGradeOrBonusState = signal<EditStateTransition>(EditStateTransition.Abort);
+    timeInvestmentState = signal<EditStateTransition>(EditStateTransition.Abort);
+    repetitionIntensityState = signal<EditStateTransition>(EditStateTransition.Abort);
+
+    constructor(
+        private alertService: AlertService,
+        private learnerProfileAPIService: LearnerProfileApiService,
+        protected translateService: TranslateService,
+    ) {}
+
+    async ngOnInit() {
+        await this.loadProfiles();
+        this.loadCourses();
+    }
+
+    courseChanged(event: Event) {
+        const val: string = (<HTMLSelectElement>event.target).value;
+        this.aimForGradeOrBonusState.set(EditStateTransition.Abort);
+        this.timeInvestmentState.set(EditStateTransition.Abort);
+        this.repetitionIntensity.set(EditStateTransition.Abort);
+
+        if (val != '-1') {
```

Comment:
```suggestion
        if (val !== '-1') {
```

Reply:
could also be wrong, but I was under the impression that typescript always prefers `===` and `!==`